Field types structure redesign (#1335)
ezsystems/developer-documentation: docs/api/field_types_reference/imageassetfield.md

# ImageAsset Field Type

> ImageAsset Field Type enables storing images in independent Content items of a generic Image content type, in the media library. It makes them reusable across system.

Reviewer: ```suggestion
ImageAsset Field Type enables storing images in independent Content items of a generic Image Content Type, in the media library. It makes them reusable across system.
```

### Input expectations

Example array:

|Type|Description|Example|
|------|------|------|
|`eZ\Publish\Core\FieldType\ImageAsset\Value`|ImageAsset Field Type value object|See below.|
|`eZ\Publish\API\Repository\Values\Content\ContentInfo`|ContentInfo instance of the Asset Content item |n/a|
|`string`| ID of the Asset Content item |`"150"`|
|`integer`| ID of the Asset Content item | `150`|

### Value object

##### Properties

Value object of `ezimageasset` contains the following properties:

| Property | Type  | Description|
|----------|-------|------------|
| `destinationContentId`  |  `int` | Related content ID. |
| `alternativeText`  |  `string` |  The alternative image text (for example "Picture of an apple."). |

``` php
// Value object content example

$imageAssetValue->destinationContentId = $contentInfo->id;
$imageAssetValue->alternativeText = "Picture of an apple.";
```

##### Constructor

> The `ImageAsset\Value` constructor will initialize a new value object with the value provided. It expects an ID of content object representing asset and the alternative text.

Reviewer: ```suggestion
The `ImageAsset\Value` constructor will initialize a new value object with the value provided. It expects an ID of a Content item representing asset and the alternative text.
```

``` php
// Constructor example

// Instantiates a ImageAsset Value object
$imageAssetValue  = new ImageAsset\Value($contentInfo->id, "Picture of an apple.");
```

### Validation

This Field Type validates if:

> - `destinationContentId` points to Content Object which has correct Content Type

Reviewer: ```suggestion
- `destinationContentId` points to a Content item which has correct Content Type
```

### Configuration

ImageAsset Field Type allows configuring the following options:

|Name|Description|Default value|
|----|-----------|-------------|

> |`content_type_identifier`|Content type used to store assets.|`image`|

Reviewer: ```suggestion
|`content_type_identifier`|Content Type used to store assets.|`image`|
```
